tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleUtil.java

```diff
@@ -31,25 +31,22 @@
 
 import org.codehaus.groovy.control.MultipleCompilationErrorsException;
 
-import org.eclipse.buildship.core.CorePlugin;
-import org.eclipse.buildship.core.configuration.BuildConfiguration;
-import org.eclipse.buildship.core.configuration.ConfigurationManager;
-import org.eclipse.buildship.core.configuration.GradleProjectNature;
-import org.eclipse.buildship.core.configuration.WorkspaceConfiguration;
-import org.eclipse.buildship.core.launch.GradleLaunchConfigurationManager;
-import org.eclipse.buildship.core.launch.GradleRunConfigurationAttributes;
-import org.eclipse.buildship.core.launch.GradleRunConfigurationDelegate;
-import org.eclipse.buildship.core.projectimport.ProjectImportConfiguration;
-import org.eclipse.buildship.core.util.binding.Validator;
-import org.eclipse.buildship.core.util.binding.Validators;
-import org.eclipse.buildship.core.util.gradle.GradleDistribution;
-import org.eclipse.buildship.core.util.gradle.GradleDistributionInfo;
-import org.eclipse.buildship.core.util.variable.ExpressionUtils;
-import org.eclipse.buildship.core.workspace.GradleBuild;
-import org.eclipse.buildship.core.workspace.GradleWorkspaceManager;
-import org.eclipse.buildship.core.workspace.NewProjectHandler;
-import org.eclipse.buildship.core.workspace.SynchronizationJob;
-import org.eclipse.buildship.core.workspace.WorkspaceOperations;
+import org.eclipse.buildship.core.BuildConfiguration;
+import org.eclipse.buildship.core.BuildConfiguration.BuildConfigurationBuilder;
+import org.eclipse.buildship.core.GradleBuild;
+import org.eclipse.buildship.core.GradleCore;
+import org.eclipse.buildship.core.GradleDistribution;
+import org.eclipse.buildship.core.GradleWorkspace;
+import org.eclipse.buildship.core.internal.CorePlugin;
+import org.eclipse.buildship.core.internal.configuration.ConfigurationManager;
```

Comment:
Do we really need all of these org.eclipse.buildship.core.internal.* packages?

Reply:
I tried to use withConnection method to execute gradle tasks like(buildCss, buildTheme..) replace our old way(invoke gradle launchConfiguration). But one problem is withConnection is a synchronized method and it will cause our watch task block other gradle process like build css and new gradle project wizard. I have to keep the old way and it still use so many buildship internal methods. I think we can move to new way while the block issue will be fixed.